src/cfn/index.ts

```diff
@@ -865,15 +867,28 @@ export async function _loadStackArgs(argsfile: string, region?: AWSRegion, profi
 
 async function stackArgsToCreateStackInput(stackArgs: StackArgs, argsFilePath: string, stackName?: string)
   : Promise<aws.CloudFormation.CreateStackInput> {
-
+  let templateLocation;
+
+  if(stackArgs.ApprovedTemplateLocation) {
+    const approvedLocation = await approvedTemplateVersionLocation(
+      stackArgs.ApprovedTemplateLocation,
+      stackArgs.Template,
+      argsFilePath
+    );
+    templateLocation = `https://s3.amazonaws.com/${approvedLocation.Bucket}/${approvedLocation.Key}`;
+  } else {
+    templateLocation = stackArgs.Template;
+  }
   // Template is optional for updates and update changesets
-  const {TemplateBody, TemplateURL} = await loadCFNTemplate(stackArgs.Template, argsFilePath);
+  const {TemplateBody, TemplateURL} = await loadCFNTemplate(templateLocation, argsFilePath);
   const {StackPolicyBody, StackPolicyURL} = await loadCFNStackPolicy(stackArgs.StackPolicy, argsFilePath);
 
   // TODO: DisableRollback
   // specify either DisableRollback or OnFailure, but not both
   const OnFailure = def('ROLLBACK', stackArgs.OnFailure)
 
+  logger.debug(`TemplateURL: ${TemplateURL}`);
```

Comment:
Let's also log whether or not the url was manipulated by `iidy` because of the approval process.